### ISSUE TITLE
Nested inheritance more than two levels produces incorrect json

### ISSUE THREAD

**Issue by AmaliaMV** [grails/grails-views#142]
I have a gson file which calls inherit to a template which call another template. Here are the templates code.

/person/index.gson
```
inherits template:"/person/index1"

json {
    attra "view index attra"
    attrb "view index attrb"
}
```

/person/_index1.gson
```
inherits template:"/person/index2"

json {
    attr1a "template index1 attra"
    attr1b "template index1 attrb"
}
```

/person/_index2.gson
```
json {
    attr2a "template index2 attr2a"
    attr2b "template index2 attr2b"
}
```

I obtained the following result for the http://localhost:8080/person/index request:
```
{{"attr2a":"template index2 attr2a","attr2b":"template index2 attr2b","attr1a":"template index1 attra","attr1b":"template index1 attrb"},"attra":"view index attra","attrb":"view index attrb"}
```

But, I would expect some like this:
```
{"attr2a":"template index2 attr2a","attr2b":"template index2 attr2b","attr1a":"template index1 attra","attr1b":"template index1 attrb","attra":"view index attra","attrb":"view index attrb"}
```

You can see in the response the list of attributes of index_2 templates and the list of attributes of index_1 template **between braces**, and next there is the list of attribute of the index templates.

You can reproduce this running the application in https://github.com/mvinas1977/sampleApp

**Comment by jameskleeh** [comment]
The example application doesn't run without a neo4j database. Please update the application without any external dependencies

**Comment by AmaliaMV** [comment]
Oh! sorry. 

I did it. Here is the new app: https://github.com/AmaliaMV/gson-view-inherits

For the request: http://localhost:8080/example/index

**Comment by olliefreeman** [comment]
Fixes #141. @AmaliaMV thank you for providing an example application so this was fixed
